src/source/canvas_source.js

```diff
@@ -22,7 +22,8 @@ import type Evented from '../util/evented';
  *        [-76.52, 39.18],
  *        [-76.52, 39.17],
  *        [-76.54, 39.17]
- *    ]
+ *    ],
+ *    contextType: '2d'
```

Comment:
If we have to make a breaking change here, maybe it's better to replace the `canvas` property with a `context` property of type `CanvasRenderingContext2D | WebGLRenderingContext`. On the other hand, that would be a property that's not JSON-serializable...

Reply:
Intuitively I think I prefer the JSON-serializable string, and feel okay with the string enum options since it must match the [string](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#Parameters) that was used to access the drawing context in the first place 🤔 though I'm happy to defer to stronger opinions…